wordpress-mobile/WordPress-iOS
# Crash when going into a like-notification details screen.

Fixes #17529 

There was an issue in how we were updating a CoreData relationship, in which we weren't properly deleting old stale managed objects, leaving them in a state that would cause a CoreData saving error.

This PR also adds unit tests to try and ensure we won't see a regression with this issue.

## Special Thanks

Special thanks to @jleandroperez for the great debugging session that resulted in us finding the actual problem.
Thanks @AliSoftware for all the help and great feedback when putting together this fix.

## To test:

- Checkout commit `e1db2d13ef8b3987a2f98408d6655e631fbea085`, which only contains the newly added unit tests.  Run them to verify that the second test crashes reproducing the issue we're seeing in production.
- Check out the latest commit from this branch, and run the unit tests again to ensure they succeed this time.

## Regression Notes

1. Potential unintended areas of impact

Notifications > Like details.  I'd expect this fix to resolve the issues and not introduce any new problem, but we have to test this thoroughly.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Added unit tests to ensure we made progress without breaking what was working.

3. What automated tests I added (or what prevented me from doing so)

I added some unit tests to ensure updates to the "likes" data works fine.

PR submission checklist:
